### PR TITLE
Feature: Command to Reboot CMS EC2 instances

### PR DESCRIPTION
**Pull request by sdford**
Adds a command to the CLI that will reboot all CMS instances one by one, ensuring that each one becomes healthy again after reboot.

### CODE REVIEW

File: src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    public boolean isRunnable(final RollingRebootWithHealthCheckCommand command) {
+
+        final StackName stackName = command.getStackName();
+        final String stackId = configStore.getStackId(stackName);
+        final Map<String, String> stackParameters = cloudFormationService.getStackParameters(stackId);
+
+        if (HEALTH_CHECK_MAP.containsKey(stackName.getName()) && stackParameters.containsKey(MIN_INSTANCES_STACK_PARAMETER_KEY)) {
+            return true;
+        }
+
```
Comment (tlisonbee) [inline src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java]: nitpick: I'd put this last clause in an else statement.  It just makes it slightly more obvious and more like http://wiki.c2.com/?SingleFunctionExitPoint

File: src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        Request requestBuilder = new Request.Builder()
+                .url(healthCheckUrl)
+                .get()
+                .build();
+
+        try {
+            final Response response = okHttpClient.newCall(requestBuilder).execute();
+            logger.debug("Health check returned status: {}, URL: {}", response.code(), healthCheckUrl);
+            int responseCode = response.code();
+            response.close();
```
Comment (tlisonbee) [inline src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java]: This close() should be in a finally block.

File: src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+import static com.nike.cerberus.command.core.RollingRebootWithHealthCheckCommand.COMMAND_NAME;
+
+/**
+ * Command to reboot the CMS cluster.
+ */
+@Parameters(commandNames = COMMAND_NAME, commandDescription = "Performs a safe rolling reboot on instances in the given cluster.")
+public class RollingRebootWithHealthCheckCommand implements Command {
+
+    public static final String COMMAND_NAME = "reboot-cluster";
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java]: Update this as we discussed in person.

File: src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import com.nike.cerberus.domain.environment.StackName;
+import com.nike.cerberus.operation.Operation;
+import com.nike.cerberus.operation.core.RollingRebootWithHealthCheckOperation;
+
+import static com.nike.cerberus.command.core.RollingRebootWithHealthCheckCommand.COMMAND_NAME;
+
+/**
+ * Command to reboot the CMS cluster.
+ */
+@Parameters(commandNames = COMMAND_NAME, commandDescription = "Performs a safe rolling reboot on instances in the given cluster.")
```
Comment (fieldju) [inline src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java]: [] Add details here that tells the user what safe is, for the help output.

File: src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java (hunk 2 above)
Comment (sdford) [inline src/main/java/com/nike/cerberus/operation/core/RollingRebootWithHealthCheckOperation.java]: Done

File: src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java (hunk 3 above)
Comment (sdford) [inline src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java]: Done

File: src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java (hunk 4 above)
Comment (sdford) [inline src/main/java/com/nike/cerberus/command/core/RollingRebootWithHealthCheckCommand.java]: Done

**Pull request merged**